Libraries/StyleSheet/NativeColorValueTypes.ios.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ * @flow strict-local
+ */
+
+'use strict';
+
+import type {ColorValue, ProcessedColorValue} from './ColorValueTypes';
+
+export opaque type NativeColorValue = {
+  semantic?: Array<string>,
+  dynamic?: {
+    light: ?(ColorValue | ProcessedColorValue),
+    dark: ?(ColorValue | ProcessedColorValue),
+  },
+};
+
+export const PlatformColor = (...names: Array<string>): ColorValue => {
+  return {semantic: names};
+};
+
+export type IOSDynamicColorTuplePrivate = {
+  light: ColorValue,
+  dark: ColorValue,
+};
+
+export const IOSDynamicColorPrivate = (
+  tuple: IOSDynamicColorTuplePrivate,
+): ColorValue => {
+  return {dynamic: {light: tuple.light, dark: tuple.dark}};
+};
+
+export const normalizeColorObject = (
+  color: NativeColorValue,
+): ?ProcessedColorValue => {
+  if ('semantic' in color) {
+    // an ios semantic color
+    return color;
+  } else if ('dynamic' in color && color.dynamic !== undefined) {
```

Comment:
Do you specifically want to have different behavior if `color.dynamic` is null here vs undefined? Otherwise `!=` vs `!==`

Reply:
true

Reply:
The analysis-bot disagrees.   it complains `eqeqeq: Expected '!==' and instead saw '!='.`.  Switching this one back.  It would only be null if someone *really* tried to bypass the IOSDynamicColor() function.

Reply:
Yeah, we don't allow `!= undefined`, only `!= null`. I think I'd still prefer `!= null`

Reply:
no problem